special-pages/pages/new-tab/app/index.js

```diff
@@ -115,14 +116,16 @@ export async function init(root, messaging, telemetry, baseEnvironment) {
                             <SettingsProvider settings={settings}>
                                 <TranslationProvider translationObject={strings} fallback={enStrings} textLength={environment.textLength}>
                                     <CustomizerProvider service={customizerApi} initialData={customizerData}>
-                                        <WidgetConfigProvider
-                                            api={widgetConfigAPI}
-                                            widgetConfigs={init.widgetConfigs}
-                                            widgets={init.widgets}
-                                            entryPoints={entryPoints}
-                                        >
-                                            <App />
-                                        </WidgetConfigProvider>
+                                        <DocumentVisibilityProvider>
```

Comment:
adding Document Visibility as a global property to prevent raceconditions with components in the tree missing it or getting a diffferent value based on when they mount.